### DISCUSSION TITLE
Image | ROCK 3C

### DISCUSSION THREAD

**Discussion by keypunch416** [MichaIng/DietPi#7058] [Image Request]
#### Formal device information
- Device name | <Rock 3C>
- Official product URL | <https://docs.radxa.com/en/rock3/rock3c>
- GitHub resource URL (if available) | <https://github.com/radxa-build/rock-3c/releases>
- Image download URL | <https://docs.radxa.com/en/rock3/rock3c/getting-started/download>
The Rock 3C variant I am using has WiFi 6/BT5.4 using a RK3566 like all other Rock 3C designs.  The other variation of the Rock 3C uses WiFi5/BT5.

#### Is the SBC officially supported by the Debian installer?
- unknown, but apt works just fine to install packages

#### If not, is a reliable 3rd party Debian image available for this SBC?
- Yes, Pine Quartz64B CLI only.
- I did not try any DE based image as I only have need for CLI only image.
- via CLI only image I saved about month ago Data/Downloads/DietPi/0902-20240318/DietPi_Quartz64B-ARMv8-Bookworm.img.xz

#### If not, are there install instructions for Debian available?
<Unknown>

There is big hope as an existing DietPi CLI only image is very close to a working Rock 3C image.  My interest is for CLI only.

Today I used a DietPi V9.2 Pine Quartz64B CLI only image that was near perfect to use.  I used the DietPi V9.2 image as what I saw today when check was not many bug changes for V9.3.  Despite imaging the V9.2 image to SD card the V9.2.1 image updated to V9.3.0.  Near perfect to use means other than inxi showing as "ARM System: Pine64 RK3566 Quartz64-B Board details: N/A serial: 427ed4ab84d91732 CPU:", no HDMI CL:I console, and only one (fine there is only one). dtb file "rk3566-quartz64-b.dtb".  i2cdetect detected a number of I2C channels which is great.  ssh worked just fine as as expected.

I do not know if the boot/uboot is such that will only load the "rk3566-quartz64-b.dtb" file by name or not.  If so could one try using a Rock 3C .dtb as test using the "rk3566-quartz64-b.dtb" file name just as test?  I do not know if the  Data/Downloads/DietPi/0902-20240318/DietPi_Quartz64B-ARMv8-Bookworm.img.xz image on an actual Pine Quartx64B HDMI the CLI would work but my hope is would.  Ideally the HDMI needs to work for CLI as very important to ebable one to see and/or correct any boot issues created by one (like me) than sometimes makes changes to files that can cause the boot to halt part way though or wants a ctrl-d to continue so one can check and/or correct the error or typo made that caused the boot to be incomplete including when systemd is not "happy'.

I installed a number of packages, such as lftp, inxi, lm-sensors, screen, sysbench, et al that all installed just fine and worked just great.  I just installed CLI based packages.  I have no intent to use GUI.

I did not try to use WiFi 6 nor BT 5.4 with the Data/Downloads/DietPi/0902-20240318/DietPi_Quartz64B-ARMv8-Bookworm.img.xz.  For my use case I will not need BT.  I will likely will use USB based WiFi adapter for distance need to connect between.  I do not think the on board PCB antenna will be sufficient for the WiFi6.  I will try the WiFi 6 at some point just to see as WiFi 6 has certain advantages.  I did not have the Rock 3C located the typical distance.  My routers are only WiFi5 at best.  I have found WiFi4 to far more stable and longer distance than WiF5.  Hence, I run my routers in WiFi4. I only use WiFi when I cannot run Ethernet cable to the networking device.  I try to use Ethernet connection rather than WiFi for any network connection and have for years.

Please note the Radxa Rock 3C CLI images is not suitable as base in my opinion let alone for months.  Radxa has not addressed the known issue Radxa has stated with current B39 Rock 3C release.  The stated issue ongoing since 10 January 2024 B39 release is `Currently there is an issue preventing the Debian CLI image from booting`  

Supporting information in attached file [Rock3C-Via-Quartz64BCLI-Image-20240503.console.txt](https://github.com/MichaIng/DietPi/files/15205800/Rock3C-Via-Quartz64BCLI-Image-20240503.console.txt)

**Comment by maxun** [comment]
Hi,
I just wanted to share my experience about this: after reading your request, I tried the DietPI Quartz64B image 9.3 on my rock-3c and it actually worked much better than I was expecting.
I got hdmi console output at first boot, I installed a few things to try. After reboot I am not seeing the console anymore on my screen however since I installed and enabled a desktop environment I was quite surprised to see that the desktop is working:

![screenshot](https://github.com/MichaIng/DietPi/assets/527928/f010256e-23f5-45fd-872e-bbf3342c2aff)

Not everything is perfect. I am getting random freeze when using the GUI and the wifi card is not recognized but after searching for an alternative os of my board I am very happy.

**Reply by MichaIng** [reply]
SoC and RAM are the same, so makes sense that it works. For other onboard features, which are not part of the SoC and differ from Quartz64B, a different device tree is required. I'll try to implement this with Armbian kernel/U-Boot sources, who added a device tree. It is community-support only, but essential things like onboard WiFi should work. If you want to test it, you can find the images here: https://github.com/armbian/community/releases (search for "rock-3c")

**Comment by maxun** [comment]
Thank you @MichaIng .
I tried the jammy desktop Armbian gnome release and it worked perfectly.
Let me know if I can help to test any image.

**Comment by keypunch416** [comment]
I did some basic initial testing of [Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz](https://github.com/armbian/community/releases/download/24.5.0-trunk.532/Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz).

Initially no HDMI console.  Suspect after reboot and/or post to manual `apt update` HDMI console worked. Great. No idea why Radxa still cannot fix B42 latest since 10 January 2024 failing to boot CLI image of their Rock3C image due to HDMI, but Radxa DE Rock 3C DE image HDMI console works per Radxa B42:

`Currently there is an issue preventing the Debian CLI image from booting, and we suggest users to use the Desktop variant instead for now. Learn more about it https://github.com/radxa-build/rock-3c/issues/5#issuecomment-1873584909.
Ubuntu Xfce: This flavor is provided as-is except for critical issues. Desktop is currently not working and we intend to fix it in a later time. This (currently broken) build is generated to catch build system issues.`

inxi shows:

`Machine:
  Type: ARM System: Radxa ROCK3 Model C details: N/A`

Correct SBC name, but unlike the DietPi Quartz64B no serial number is shown.  Is this related the Device Tree? 

Only one I2C bus with Armbian Rock 3C , Bus 0 
[Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-i2cdetect.console.txt](https://github.com/MichaIng/DietPi/files/15210977/Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-i2cdetect.console.txt)
.  Compared to many I2C buses of the DietPi Quartz64B 
[DietPi_Quartz64B-ARMv8-Bookworm.img.xz-v9.2.1UpdateTo9.3.0-minimal-i2cdetect.console.txt](https://github.com/MichaIng/DietPi/files/15210987/DietPi_Quartz64B-ARMv8-Bookworm.img.xz-v9.2.1UpdateTo9.3.0-minimal-i2cdetect.console.txt)
.  That is not a negative.  It makes more sense to not configure all I2C buses when one may want to use those pins for other purposes such as, one or more specif I2C buses SPI, general GPIO use, er al.  I know the Radxa images have basic .dtb with many .dto overlays one can pick and choose for ones specific needs.  I would think it be good for DietPi to include all these .dto files that Radxa has as source available via github place for .dtb and .dto sources for all chips sets.

gpioinfo which I forgot to run with the DietPi Quartz64B image shows some GPIO pins assigned 
[Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-gpioinfo.console.txt](https://github.com/MichaIng/DietPi/files/15210949/Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-gpioinfo.console.txt),
but clearly missing the GPIO names I know show in the Radxa 4B, 3A, and 5B images.

No WiFi available 
[Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-WiFi.console.txt](https://github.com/MichaIng/DietPi/files/15211023/Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal.img.xz-WiFi.console.txt) perhaps as I have Rock 3C with WiFi 6 and BT 5.4.  As previous noted there are two versions of Rock 3C one with WiFi 6/BT 5.4 and one with WiFi 5/BT 5.  Perhaps as [user maxum working WiFi](https://github.com/MichaIng/DietPi/discussions/7058#discussioncomment-9313483) is due to his Rock 3C having WiFi 5/BT 5?

I made no attempt to test BT 5.4 on my Rock 3C with WiFi 6/BT 5.4.

I did not mention in my initial request, but will now.  I did try the Armbian Pine Quartz64B image, but not able to boot from.  There were many .dtb files for various RK3566 SBCs, but no Rock3c .dtb existed.  I assume, possibly incorrectly, that the current Armbian Pine Quartz64B image I tried [Armbian_community_24.5.0-trunk.532_Quartz64b_bookworm_current_6.1.90_minimal.img.xz](https://github.com/armbian/community/releases/download/24.5.0-trunk.532/Armbian_community_24.5.0-trunk.532_Quartz64b_bookworm_current_6.1.90_minimal.img.xz) yesterday before trying a DietPi version did not boot as the boot looked for a Rock3c .dtb file which did not exist rather than load the Quartz64B .dtb.

Lost all my prior DietPi console took bit from for initial request and lost some key parts of console of Armbian Rock 3C image testing due to initially annoying manner that first Armbian root logon does I cannot control and annoying .bash_logout for non-root user!  Really really annoying and makes it near impossible to actually ensure have what need to report issues.  Armbian has some bugs due to Armbian specific OS configuration that causes serious errors in `apt update` process no matter if use `sudo` when logged in as non-root user or logged in as root.

I did not report these issues that are clearly not Rock 3C image specific (seen such issues with different SBC Armbian images) as the whole report an issue to Armbian has so many checkpoints and such that in essence impossible to report Armbian issues, ergo why such basic Armbian bugs exist and exist for long time.  I like the Diet Pi process to report issues or even request new image as has template when open as well as allowing one to explain and provide the needed details.  With all the ins/outs/checkpoints and inability to actually report an issue to Armbian I do not despite my deep technical skills and background to figure out how to do so with Armbian assuming I do not have to make many many attempts before I find a successful way to do so with the Armbian bug reporting process.

I am looking forward to when DietPi has a Rock 3C image.  I find the DietPi images are more suitable for my use case.  As FYI, I do feel there is one very annoying aspect of recent DietPi design, not a bug, I do not like at all.  That is for different issue when I decide to open such an issue.

**Comment by keypunch416** [comment]
I did a test yesterday of a Radxa 20240109 B41 release I discovered way down in list of releases after releases earlier, that had many issues despite no issues listed in release notes to point of never actually booting for CLI.  Today I tried the B42 XFCE release that provided some interesting information, but I will need to repeat the tests as post update items that worked, did not post update.  A question I have is would it be better for me to post results of image tests, or results of Device Tree, GPIO, Wireless, et al for each image as a single post or post results of the image test?  BTW I have learned some things along way in commands need to run to document or better document results.

**Comment by keypunch416** [comment]
So far I have the following commands to obtain information from testing related images:

`uname -a
inxi -F

lscpu
lsusb
lspci
lspci -k
lsmod
top -n1
ps aux
free

sudo ifconfig
sudo ip addr
echo "iw dev wlan0 scan"
sudo iw dev wlan0 scan

ls -lsR --full-time /boot/*
ls /proc/device-tree
dtc --version
dtc -I fs /sys/firmware/devicetree/base

ls -ls --full-time /dev/i2c*
ls -ls --full-time /dev/spi*
sudo i2cdetect -y 0
sudo i2cdetect -y 1
sudo i2cdetect -y 2
sudo i2cdetect -y 3
sudo i2cdetect -y 4
sudo i2cdetect -y 5
sudo i2cdetect -y 6
sudo i2cdetect -y 7
sudo i2cdetect -y 8
sudo i2cdetect -y 9

sudo gpioinfo`

Are there any other commands I need to run to obtain information useful to understand the common and differences of the related images?

I would suggest the related images be Quartz64B from DietPi, Rock 3C images from Armbian and Radxa.  I do not think there is a need to test the Armbian Quartz64B, but if there is a different opinion let me know.

**Comment by maxun** [comment]
Here the output for Armbian_community_24.5.0-trunk.532_Rock-3c_bookworm_current_6.6.30_minimal:

[armbian.txt](https://github.com/MichaIng/DietPi/files/15227056/armbian.txt)

**Comment by maxun** [comment]
Radxa ROCK 3C support is now in the mainline kernel:

https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts

**Comment by keypunch416** [comment]
As FYI I have been using the Armbian Rock 3C image as at about 2+ months ago.  The reason for using the Armbian Rock 3C image was there were to many bugs/issues with the current Raspberry Pi image for RPi2.  The Armbian image did not have the "standard" I2C pins defined via image DTB.  After some effort I was able to find the correct DTC image to compile DTBO overlay that was then added to /boot/dietpiEnv.txt overlays=.  The Rock 3C runs 24/7 365 logging wide variety of weather data such as temperature of for over 20 temperature sensors, humidity, barometric pressure, various light intensity sensors, rainfall, lightning, et al every 15 seconds.  Until about 18 months ago this was being done with a RPi1B for years without any OS stability issues.  I changed to RPi2 as lower power/current requirement than RPiBR2.

I tried to find the SPIDEV overlay, but to date I have not found the correct source for SPIDEV for Rock 3C.  The weather system SBC does not need SPI.  The ADCs I want to use for seismic data logging are SPI based ADCs.  I do not like the root requirement for Raspberry OS to use SPI via BCM2835 Library.  Seismic data recording requirements are 200 samples per second per instrument.  That is about 20Gb data per day per instrument.

It took about 2 weeks to have stable Rock 3C not due to any Armbian issue.  The stability issue was clearly Debian changes made to Bookworm (12.x) release that never existed before in any release of Debian prior to Bookworm (12.x) causing all kinds of problems with USB /dev/ttyUSBxx assignment changes.  It took me alot of time via trial and error experimenting to undo this mess of changes related to USB /dev/ttyUSBxx stability.  I have not reported the issue as prior experience in doing so only results in much churns and telling em I do not know what I am talking about.  Again this /dev/ttyUSBxx assignment stability issues is Debian and not Armbian, hence I would expect any OS based on Bookworm (12.x) will have same /dev/ttyUSBxx stability issues.

So far apt updates works fine now.  In early weeks of my starting to use Rock 3C the apt update for kernel had some clearly Armbian bugs not specific to Rock 3C that have since been resolved.  The bugs were such that the Kernel would not actually be updated due to errors in the Armbian Kernel package install scripts.

All the code I have written for my weather data logging compiled just fine on Rock 3C image with no issues as has been case for years.

Sadly the GPIO pins are not defined as per the RK3566 SoC.  Hence programming the GPIO is still a serious gap for the Rock 3C image as well as many other Armbian/DietPi SBC images from my experience.

I am very happy with the low power, lower than RPi2, of Rock 3C and very very happy with the processing power of RK3566 and the cores of RK3566.

**Comment by keypunch416** [comment]
I am closing this request as I no longer wish to use DietPi on any system.  Yet again DietPi has trashed the DietPI system in multiple ways.  These have occurred on the SBCs that are not Rock 3C as of course there is no DietPi for Rock 3C.  The instance that last occurred is last few days is such that now impossible to recover from.  This is not media corruption.  Clearly DietPi can easily make a mess of basic OS elements that not even someone like myself with deep and broad technical experience and skills could not dig out of after 48+ man hours of effort.  I have been using Linux as my only OS since 1998 (Kernel 2.2 era) For my multiple 24/7/365 including seismic recording I cannot trust DietPi to provide a stable (longer than few months) OS for these 24/7/365 systems that are all only CLI based, I.E. no DE/Desktop of any type installed.  For the other 2 remaining 24/7/365 systems I have on DietPi I will move those off of DietPi in new few months when time allows me to create and test the migration off these remaining DietPi systems.  Please do not ask what when wrong as list is too long.  I do not have any interest nor the time required to document, retest, and prove the multiple issues that trashed DietPi (not media) nor the many other issues that are likely related that cause alot of time and effort to work around.  I know the effort and process required as for over 30 years my job was to test and/or diagnose (using test equipment/multiple software debuggers) very technical, complex technology using multiple OS/Hardware/Software/Networks/Internet/ISPs/Internet Backbone/Hosting/Provisioning/Banking/Huge Databases/Firmware/Microcode/et al often spanning North America.  I clearly know the processes, time required, and demands technically reporting issues involves very very very well.

**Reply by MichaIng** [reply]
> Please do not ask what when wrong as list is too long.

This is very sad to read. Nothing is perfect, but we have somewhere above 100.000 users from which most do not run into a long list of issues. Quite likely that it was one single issue which caused a rats tail? And sharing the initial issue with us does not only give us a chance to in case enhance DietPi, but also gives us a chance to help you identifying and fixing it early, before you wasted 48+ hours.

As you know there are many layers at which things can go wrong, and DietPi itself provides only the thin layer of some setup scripts and configs at the top, at best fixing a few things broken in Debian packages, Armbian build system, or underlying kernel/bootloader sources. Since you do not want to share any info, I don't know whether the issue you faced is hence caused by DietPi at all, or whether it is something you can run into with any next system. I just hope you do not blame and give up on DietPi for such an issue.

In any case, good luck, and hopefully a better experience with the next OS you try.
